onchain/permissionless-arbitration/offchain/player/state.lua

```diff
@@ -0,0 +1,119 @@
+local constants = require "constants"
+local bint = require 'utils.bint' (256) -- use 256 bits integers
+
+local Client = require "blockchain.client"
+
+local Player = {}
+Player.__index = Player
+
+function Player:new(root_tournament_address, player_index, commitment_builder, machine_path)
```

Comment:
I think this type should be called `State`. We can make it leaner I believe.

Reply:
I don't think we need the `player_index`, I think it's just used for sending tx. Can you verify? Maybe it makes sense to split the client into reader/sender.

Reply:
I think we first need to define a pair of types: tournament and match.

Tournament can be constructed from its address and optional parent match. It has:
* address;
* optional pointer to its parent match;
* level;
* list of commitments made at that tournament;
* list of matches.

Match can be constructed from its tournament. It has:
* tournament it's in;
* commitment one and two;
* current left, right, height, and running leaf;
* leaf_cycle and base_big_cycle;
* optional child tournament.

With that, I believe we can build the whole "tree" of tournament/match. It mirrors the onchain structure. Just by passing the address of the root tournament, the whole structure is built. I think the main difference is that the current code only fetches the matches/tournaments the player is in. The suggestion fetches everything. We'll need that for the simplified clock.

We also remove dependency on machine and commitment builder.

The next part is the strategy, which may need to "search" its commitments on this tournament/match structure. So it may be a good idea to add search/find methods. Like, `tournament:find_match_with_commitment(c)` or something.

Reply:
Makes sense. I'll work on this change.

Reply:
I think we'll need a new event when someone joins the tournament. @GCdePaula what do you think?

Reply:
Agreed! Although I think we should make it `event commitmentJoined(Tree.Node right, Tree.Node left);`. It's possible we may even remove the `matchCreated` event, since we can derive that from the `commitmentJoined` event (every pair of consecutive `matchCreated` implies a `commitmentJoined`). But let's leave this second step for later, unless you wanna do it now.

Reply:
I created an issue to track this topic.
https://github.com/cartesi/rollups/issues/242